Wrappers/Python/cil/framework/acquisition_data.py

```diff
@@ -80,20 +81,18 @@ def __init__(self,
                 raise ValueError("Deprecated: 'dimension_labels' cannot be set with 'allocate()'. Use 'geometry.set_labels()' to modify the geometry before using allocate.")
 
         if array is None:
-            if dtype is None:
-                dtype = geometry.dtype
-            array = numpy.empty(geometry.shape, dtype)
-    
+            xp = array_api_compat.numpy
+            dtype = kwargs.get('dtype', xp.float32)
```

Comment:
did you intend to change the default behaviour from `dtype = geometry.dtype` to `xp.float32`?